samza-core/src/main/scala/org/apache/samza/job/JobRunner.scala

```diff
@@ -161,10 +167,8 @@ class JobRunner(config: Config) extends Logging {
   }
 
   def status(): ApplicationStatus = {
-    val jobFactory: StreamJobFactory = getJobFactory
-
     // Create the actual job, and get its status.
-    jobFactory.getJob(config).getStatus
+    streamJob.getStatus
```

Comment:
Same issue here. "streamJob" may not survive between commands, so this isn't safe. 

e.g. this will definitely break the deployment at LinkedIn

Reply:
Good point. I realized that this will break the existing deployment if the streamJob is YarnJob. I have updated the patch to fix it. Can you take another look?